premake5.lua

```diff
@@ -662,3 +662,71 @@ if (os.istarget("linux")) then
     configuration { "Release" }
     targetdir "target/app/Release"
 end
+
+-- HEADLESS APP
+
+project "surge-headless"
+kind "ConsoleApp"
+
+defines
+{
+    "TARGET_HEADLESS=1"
+}
+
+plugincommon()
+
+files {
+    "src/headless/main.cpp",
+    "src/headless/DisplayInfoHeadless.cpp",
+    "src/headless/UserInteractionsHeadless.cpp",
+    "src/headless/LinkFixesHeadless.cpp"
+    }
+
+excludes {
+    "src/common/gui/*"
+}
+
+includedirs {
+    "src/headless"
+}
+
+configuration { "Debug" }
+targetdir "target/headless/Debug"
+targetsuffix "-Debug"
+
+configuration { "Release" }
+targetdir "target/headless/Release"
+
+configuration {}
+
+if (os.istarget("macosx")) then
+       excludes{
+           VSTGUI .. "vstgui_mac.mm",
+           VSTGUI .. "vstgui_uidescription_mac.mm",
+           "src/mac/DisplayInfoMac.mm",
+           "src/mac/UserInteractionsMac.cpp",
+       }
+end
+
+if (os.istarget("windows")) then
+       excludes{
+           VSTGUI .. "vstgui_win32.cpp",
+           VSTGUI .. "vstgui_uidescription_win32.cpp",
+           "src/windows/DisplayInfoWin.cpp",
+           "src/windows/UserInteractionsWin.cpp",
+       }
+end
+
+if (os.istarget("linux")) then
+    excludes {
+            VSTGUI .. "vstgui.cpp",
+            VSTGUI .. "lib/platform/linux/**.cpp",
+            VSTGUI .. "lib/platform/common/genericoptionmenu.cpp",
+            VSTGUI .. "lib/platform/common/generictextedit.cpp",
+           "src/linux/DisplayInfoLinux.cpp",
+           "src/linux/UserInteractionsLinux.cpp",
+    }
+end
+
+
+
```

Comment:
Three trailing new line characters.

Reply:
And in some places it looks as if you were indenting more than four spaces. This diff needs a throughout clean up.